scripts/check-imports.go

```diff
@@ -69,18 +73,36 @@ func main() {
 	}
 
 	sort.Slice(pkgs, func(i, k int) bool { return pkgs[i].ID < pkgs[k].ID })
+	correct := true
 	for _, pkg := range pkgs {
-		process(pkg)
+		if !correctPackage(pkg) {
+			correct = false
+		}
+	}
+
+	if !correct {
+		fmt.Fprintln(os.Stderr, "imports not in the correct order")
+		os.Exit(1)
 	}
 }
 
-func process(pkg *packages.Package) {
+func correctPackage(pkg *packages.Package) bool {
+	correct := true
 	for i, file := range pkg.Syntax {
-		checkImports(pkg.Fset, pkg.CompiledGoFiles[i], file)
+		path := pkg.CompiledGoFiles[i]
+		if !correctImports(pkg.Fset, path, file) {
+			if !isGenerated(path) { // ignore generated files
```

Comment:
Should we ignore generated files before doing the actual check for correct imports?

Reply:
OK